app/src/main/java/com/metro/gwuexplorer/AlertManager.kt

```diff
@@ -0,0 +1,70 @@
+package com.metro.gwuexplorer
+
+import android.widget.Toast
+import okhttp3.*
+import okhttp3.logging.HttpLoggingInterceptor
+import org.json.JSONObject
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+
+class AlertManager {
+    private val okHttpClient: OkHttpClient
+
+    init {
+        val builder = OkHttpClient.Builder()
+
+
+        builder.connectTimeout(20, TimeUnit.SECONDS)
+        builder.readTimeout(20, TimeUnit.SECONDS)
+        builder.writeTimeout(20, TimeUnit.SECONDS)
+
+        val logging = HttpLoggingInterceptor()
+        logging.level = HttpLoggingInterceptor.Level.BODY
+        builder.addInterceptor(logging)
+        okHttpClient = builder.build()
+    }
+
+
+
+    fun retrieveAlert(
+        successCallback: (List<Alert>) -> Unit,
+        errorCallback: (Exception) -> Unit
+    ) {
+        val primaryKey = "507c1527b662401aaee3f16396982ccc"
```

Comment:
Make sure you hide this...

This is also a public repo, so I'd also suggest regenerating your key, just in case...

At the end of Lecture 7, I have some slides with ideas of how you can hide your metro key